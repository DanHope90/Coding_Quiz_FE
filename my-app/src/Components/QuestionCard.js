/* eslint-disable jsx-a11y/control-has-associated-label */
/* eslint-disable no-console */
import React, { useState } from "react";
import PropTypes from "prop-types";
import "../Styles/QuestionCard.css";

function QuestionCard(props) {
  const {
    question, score, setScore, qNumber, setQNumber,
  } = props;
  console.log("CurrentScore:", score);
  console.log("CurrentQNumber", qNumber);
  console.log(question.correct_answer);

  const [chosenAnswer, setChosenAnswer] = useState("");

  // Below checks if the answer chosen by the user matches the
  // actual correct answer. If so, increments score by 1,
  // otherwise keeps it as it is. Also, the qNumber is increased
  // by 1 when an answer (correct or not) is selected.

  function checkAnswer() {
    if ((chosenAnswer === question.correct_answer)) {
      setScore(score + 1);
      setQNumber(qNumber + 1);
    } else if ((chosenAnswer === "")) {
      setScore(score);
      setQNumber(qNumber);
    } else if ((chosenAnswer !== question.correct_answer)) {
      setScore(score);
      setQNumber(qNumber + 1);
    }
  }

  checkAnswer(chosenAnswer);
  console.log(chosenAnswer);

  return (
    <div>
      <div>
        {question.question}
      </div>
      <div className="answers">
        {/* When the buttons are pressed, they seem to remain pressed until a different one is
        pressed.So the way it is rendering now may allow the user to accrue an infinite score if
        they wanted. So the question needs to disappear after its been answered to stop the user
        from answering it multiple times.  */}
        <button type="submit" onClick={() => setChosenAnswer(question.answers[0].a)}>{question.answers[0].a}</button>
        <button type="submit" onClick={() => setChosenAnswer(question.answers[1].b)}>{question.answers[1].b}</button>
        <button type="submit" onClick={() => setChosenAnswer(question.answers[2].c)}>{question.answers[2].c}</button>
        {/* Fourth button still looping for some reason */}
        <button type="submit" onClick={() => setChosenAnswer(question.answers[3].d)}>{question.answers[3].d}</button>
      </div>
    </div>
  );
}

export default QuestionCard;

QuestionCard.propTypes = {
  question: PropTypes.string,
  setScore: PropTypes.func,
  score: PropTypes.number,
  qNumber: PropTypes.number,
  setQNumber: PropTypes.func,
}.isRequired;
