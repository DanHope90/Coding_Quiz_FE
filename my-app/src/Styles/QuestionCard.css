.answers{
    display: flex;
    flex-flow: wrap;
    padding: 10px;
    margin: 10px;
    gap: 10px;
    align-items: center;
};

.answer-button {
    color: white;
    background-color: black;
};

.answer-button:active {
    color:red;
    background-color: white;
}