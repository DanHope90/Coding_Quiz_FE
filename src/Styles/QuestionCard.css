.question-body {
    display: flex;
    flex-flow: wrap;
    align-items: center;
    justify-content: center;
}

.question-text {
    padding: 10px;
    font-size: 16px;
}

.answers{
    display: flex;
    flex-flow: wrap;
    padding: 10px;
    margin: 10px;
    gap: 10px;
    align-items: center;
    justify-content: center;
};

.answer-button {
    color: white;
    background-color: black;
};

.answer-button:active {
    color:red;
    background-color: white;
}

@media only screen and (max-width: 650px) {
    .question-text{
        font-size: 12px;
    }
}