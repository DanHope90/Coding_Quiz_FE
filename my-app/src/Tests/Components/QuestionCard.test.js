import React from "react";
import renderer from "react-test-renderer";
import QuestionCard from "../../Components/QuestionCard";

test("renders the Question Card component correctly", () => {
  const dummyQuestion = {
    question: "What is the symbol for multiplcation?",
    answers: [{ a: "*" }, { b: "/" }, { c: "-" }, { d: "+" }],
    correct_answer: "*",
  };

  const testRenderer = renderer.create(<QuestionCard value={dummyQuestion} />);
  expect(testRenderer).toMatchSnapshot();
});
