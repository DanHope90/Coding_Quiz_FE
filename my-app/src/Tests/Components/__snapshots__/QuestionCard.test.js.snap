// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Question Card component correctly 1`] = `
<div>
  <div>
    What is the symbol for multiplication?
  </div>
  <div
    className="answers"
  >
    <div>
      <button
        className="answer-button"
        onClick={[Function]}
        type="submit"
      >
        *
      </button>
    </div>
    <div>
      <button
        className="answer-button"
        onClick={[Function]}
        type="submit"
      >
        /
      </button>
    </div>
    <div>
      <button
        className="answer-button"
        onClick={[Function]}
        type="submit"
      >
        -
      </button>
    </div>
    <div>
      <button
        className="answer-button"
        onClick={[Function]}
        type="submit"
      >
        +
      </button>
    </div>
  </div>
</div>
`;
